[git_stats]: fatal: could not read Username for 'https://github.com': No such device or address

```
2017-08-25 14:27:28,220 [INFO] celery.worker.strategy: Received task: selinon.SelinonTaskEnvelope[c5cfc9e7-97e3-47e2-b63d-37a36a468b8c]  
{"task_id": "c5cfc9e7-97e3-47e2-b63d-37a36a468b8c", "flow_name": "bayesianFlow", "queue": "STAGE_ingestion_FinalizeTaskError_v0", "task_name": "FinalizeTaskError", "event": "TASK_START", "node_args": {"force_graph_sync": true, "_audit": {"started_at": "2017-08-25T14:23:44.928238", "version": "v1", "ended_at": "2017-08-25T14:23:45.592632"}, "version": "0.2.4", "_release": "pypi:clojure__py:0.2.4", "recursive_limit": 0, "document_id": 240915, "force": true, "name": "clojure__py", "ecosystem": "pypi"}, "dispatcher_id": "8280cb4f-3837-4738-bc8b-be733aee526d", "parent": null}
{"task_id": "c5cfc9e7-97e3-47e2-b63d-37a36a468b8c", "event": "TASK_END", "parent": null, "flow_name": "bayesianFlow", "queue": "STAGE_ingestion_FinalizeTaskError_v0", "task_name": "FinalizeTaskError", "storage": "BayesianPostgres", "node_args": {"force_graph_sync": true, "_audit": {"started_at": "2017-08-25T14:23:44.928238", "version": "v1", "ended_at": "2017-08-25T14:23:45.592632"}, "version": "0.2.4", "_release": "pypi:clojure__py:0.2.4", "recursive_limit": 0, "document_id": 240915, "force": true, "name": "clojure__py", "ecosystem": "pypi"}, "dispatcher_id": "8280cb4f-3837-4738-bc8b-be733aee526d"}
2017-08-25 14:27:28,311 [INFO] celery.app.trace: Task selinon.SelinonTaskEnvelope[c5cfc9e7-97e3-47e2-b63d-37a36a468b8c] succeeded in 0.09048445100779645s: None
2017-08-25 14:27:47,892 [INFO] celery.worker.strategy: Received task: selinon.SelinonTaskEnvelope[b4d16dec-8c85-477d-8ed9-54646e0c64c7]  
{"task_id": "b4d16dec-8c85-477d-8ed9-54646e0c64c7", "flow_name": "bayesianPackageAnalysisFlow", "queue": "STAGE_ingestion_git_stats_v0", "task_name": "git_stats", "event": "TASK_START", "node_args": {"force_graph_sync": true, "_audit": {"started_at": "2017-08-25T14:27:26.563165", "version": "v1", "ended_at": "2017-08-25T14:27:26.593944"}, "name": "clojure__py", "url": "https://github.com/halgari/clojure-py", "_release": "pypi:clojure__py:None", "recursive_limit": 0, "document_id": 808, "force": true, "ecosystem": "pypi"}, "dispatcher_id": "03c94717-ea19-48ae-b2e8-a0984e652669", "parent": {}}
2017-08-25 14:27:47,950 [WARNING] f8a_worker.utils: command ['git', 'config', '--get', 'user.name'] ended with 1

2017-08-25 14:27:47,964 [WARNING] f8a_worker.utils: command ['git', 'config', '--get', 'user.email'] ended with 1

fatal: could not read Username for 'https://github.com': No such device or address
2017-08-25 14:27:48,177 [WARNING] f8a_worker.utils: command ['git', 'clone', 'https://github.com/halgari/clojure-py', '/tmp/tmp8bioyloc'] ended with 128
Cloning into '/tmp/tmp8bioyloc'...

2017-08-25 14:27:48,178 [ERROR] f8a_worker.utils: exception is fatal
{"task_id": "b4d16dec-8c85-477d-8ed9-54646e0c64c7", "event": "TASK_FAILURE", "retried_count": 0, "parent": {}, "flow_name": "bayesianPackageAnalysisFlow", "queue": "STAGE_ingestion_git_stats_v0", "task_name": "git_stats", "node_args": {"force_graph_sync": true, "_audit": {"started_at": "2017-08-25T14:27:26.563165", "version": "v1", "ended_at": "2017-08-25T14:27:26.593944"}, "name": "clojure__py", "url": "https://github.com/halgari/clojure-py", "_release": "pypi:clojure__py:None", "recursive_limit": 0, "document_id": 808, "force": true, "ecosystem": "pypi"}, "dispatcher_id": "03c94717-ea19-48ae-b2e8-a0984e652669", "what": "Traceback (most recent call last):\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/utils.py\", line 255, in get_command_output\n    out = check_output(args, universal_newlines=True, **kwargs)\n  File \"/usr/lib64/python3.4/subprocess.py\", line 617, in check_output\n    raise CalledProcessError(retcode, process.args, output=output)\nsubprocess.CalledProcessError: Command '['git', 'clone', 'https://github.com/halgari/clojure-py', '/tmp/tmp8bioyloc']' returned non-zero exit status 128\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/utils.py\", line 143, in tempdir\n    yield dirpath\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/workers/git_stats.py\", line 26, in _get_log\n    git = Git.clone(url, tmp_dir)\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/process.py\", line 67, in clone\n    TimedCommand.get_command_output(cmd, graceful=False)\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/utils.py\", line 239, in get_command_output\n    return get_command_output(args, graceful, is_json, **kwargs)\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/utils.py\", line 265, in get_command_output\n    raise TaskError(\"Error during running command %s: %r\" % (args, ex.output))\nf8a_worker.errors.TaskError: Error during running command ['git', 'clone', 'https://github.com/halgari/clojure-py', '/tmp/tmp8bioyloc']: \"Cloning into '/tmp/tmp8bioyloc'...\\n\"\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/lib/python3.4/site-packages/selinon/selinonTaskEnvelope.py\", line 115, in run\n    result = task.run(node_args)\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/base.py\", line 38, in run\n    result = self.execute(node_args)\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/workers/git_stats.py\", line 132, in execute\n    master_log = self._get_log(arguments['url'])\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/workers/git_stats.py\", line 29, in _get_log\n    log = git.log()\n  File \"/usr/lib64/python3.4/contextlib.py\", line 77, in __exit__\n    self.gen.throw(type, value, traceback)\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/utils.py\", line 145, in tempdir\n    shutil.rmtree(dirpath)\n  File \"/usr/lib64/python3.4/shutil.py\", line 459, in rmtree\n    onerror(os.lstat, path, sys.exc_info())\n  File \"/usr/lib64/python3.4/shutil.py\", line 457, in rmtree\n    orig_st = os.lstat(path)\nFileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp8bioyloc'\n"}
{"task_id": "b4d16dec-8c85-477d-8ed9-54646e0c64c7", "error_type": "<class 'FileNotFoundError'>", "event": "STORAGE_STORE_ERROR", "storage_task_name": "git_stats", "error_value": "[Errno 2] No such file or directory: '/tmp/tmp8bioyloc'", "flow_name": "bayesianPackageAnalysisFlow", "task_name": "git_stats", "error_traceback": "  File \"/usr/lib/python3.4/site-packages/selinon/selinonTaskEnvelope.py\", line 115, in run\n    result = task.run(node_args)\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/base.py\", line 38, in run\n    result = self.execute(node_args)\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/workers/git_stats.py\", line 132, in execute\n    master_log = self._get_log(arguments['url'])\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/workers/git_stats.py\", line 29, in _get_log\n    log = git.log()\n  File \"/usr/lib64/python3.4/contextlib.py\", line 77, in __exit__\n    self.gen.throw(type, value, traceback)\n  File \"/usr/lib/python3.4/site-packages/f8a_worker/utils.py\", line 145, in tempdir\n    shutil.rmtree(dirpath)\n  File \"/usr/lib64/python3.4/shutil.py\", line 459, in rmtree\n    onerror(os.lstat, path, sys.exc_info())\n  File \"/usr/lib64/python3.4/shutil.py\", line 457, in rmtree\n    orig_st = os.lstat(path)\n", "node_args": {"force_graph_sync": true, "_audit": {"started_at": "2017-08-25T14:27:26.563165", "version": "v1", "ended_at": "2017-08-25T14:27:26.593944"}, "name": "clojure__py", "url": "https://github.com/halgari/clojure-py", "_release": "pypi:clojure__py:None", "recursive_limit": 0, "document_id": 808, "force": true, "ecosystem": "pypi"}, "storage_name": "PackagePostgres", "record_id": null}
2017-08-25 14:27:48,255 [ERROR] celery.app.trace: Task selinon.SelinonTaskEnvelope[b4d16dec-8c85-477d-8ed9-54646e0c64c7] raised unexpected: FileNotFoundError(2, 'No such file or directory')
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/f8a_worker/utils.py", line 255, in get_command_output
    out = check_output(args, universal_newlines=True, **kwargs)
  File "/usr/lib64/python3.4/subprocess.py", line 617, in check_output
    raise CalledProcessError(retcode, process.args, output=output)
subprocess.CalledProcessError: Command '['git', 'clone', 'https://github.com/halgari/clojure-py', '/tmp/tmp8bioyloc']' returned non-zero exit status 128

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/f8a_worker/utils.py", line 143, in tempdir
    yield dirpath
  File "/usr/lib/python3.4/site-packages/f8a_worker/workers/git_stats.py", line 26, in _get_log
    git = Git.clone(url, tmp_dir)
  File "/usr/lib/python3.4/site-packages/f8a_worker/process.py", line 67, in clone
    TimedCommand.get_command_output(cmd, graceful=False)
  File "/usr/lib/python3.4/site-packages/f8a_worker/utils.py", line 239, in get_command_output
    return get_command_output(args, graceful, is_json, **kwargs)
  File "/usr/lib/python3.4/site-packages/f8a_worker/utils.py", line 265, in get_command_output
    raise TaskError("Error during running command %s: %r" % (args, ex.output))
f8a_worker.errors.TaskError: Error during running command ['git', 'clone', 'https://github.com/halgari/clojure-py', '/tmp/tmp8bioyloc']: "Cloning into '/tmp/tmp8bioyloc'...\n"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/celery/app/trace.py", line 367, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/lib/python3.4/site-packages/celery/app/trace.py", line 622, in __protected_call__
    return self.run(*args, **kwargs)
  File "/usr/lib/python3.4/site-packages/selinon/selinonTaskEnvelope.py", line 170, in run
    raise self.retry(max_retries=0, exc=exc)
  File "/usr/lib/python3.4/site-packages/celery/app/task.py", line 668, in retry
    raise_with_context(exc)
  File "/usr/lib/python3.4/site-packages/selinon/selinonTaskEnvelope.py", line 115, in run
    result = task.run(node_args)
  File "/usr/lib/python3.4/site-packages/f8a_worker/base.py", line 38, in run
    result = self.execute(node_args)
  File "/usr/lib/python3.4/site-packages/f8a_worker/workers/git_stats.py", line 132, in execute
    master_log = self._get_log(arguments['url'])
  File "/usr/lib/python3.4/site-packages/f8a_worker/workers/git_stats.py", line 29, in _get_log
    log = git.log()
  File "/usr/lib64/python3.4/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/lib/python3.4/site-packages/f8a_worker/utils.py", line 145, in tempdir
    shutil.rmtree(dirpath)
  File "/usr/lib64/python3.4/shutil.py", line 459, in rmtree
    onerror(os.lstat, path, sys.exc_info())
  File "/usr/lib64/python3.4/shutil.py", line 457, in rmtree
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp8bioyloc'
```

--- a/f8a_worker/process.py
+++ b/f8a_worker/process.py
@@ -44,6 +44,9 @@ class Git(object):
             TimedCommand.get_command_output(["git", "config", "--global", "user.name", user_name])
         if not TimedCommand.get_command_output(["git", "config", "--get", "user.email"]):
             TimedCommand.get_command_output(["git", "config", "--global", "user.email", user_email])
+        # Use 'true' as external program to ask for credentials, i.e. don't ask
+        # Better would be GIT_TERMINAL_PROMPT=0, but that requires git >= 2.3
+        TimedCommand.get_command_output(["git", "config", "--global", "core.askpass", "/usr/bin/true"])
 
     @classmethod
     def clone(cls, url, path, depth=None, branch=None):

--- a/f8a_worker/process.py
+++ b/f8a_worker/process.py
@@ -15,6 +15,7 @@ from urllib.parse import urljoin, urlparse
 
 from f8a_worker.conf import get_configuration
 from f8a_worker.enums import EcosystemBackend
+from f8a_worker.errors import TaskError
 from f8a_worker.utils import cwd, TimedCommand, compute_digest, MavenCoordinates, url2git_repo
 
 logger = logging.getLogger(__name__)
@@ -59,6 +60,7 @@ class Git(object):
         :param branch: str
         :return: instance of Git()
         """
+        orig_url = url
         cls.config()
         # git clone doesn't understand urls starting with: git+ssh, git+http, git+https
         url = url2git_repo(url)
@@ -67,7 +69,10 @@ class Git(object):
             cmd.extend(["--depth", depth])
         if branch is not None:
             cmd.extend(["--branch", branch])
-        TimedCommand.get_command_output(cmd, graceful=False)
+        try:
+            TimedCommand.get_command_output(cmd, graceful=False)
+        except TaskError as exc:
+            raise TaskError("Unable to clone: %s" % orig_url) from exc
         return cls(path=path)
 
     @classmethod

--- a/f8a_worker/utils.py
+++ b/f8a_worker/utils.py
@@ -151,7 +151,8 @@ def tempdir():
     try:
         yield dirpath
     finally:
-        shutil.rmtree(dirpath)
+        if os_path.isdir(dirpath):
+            shutil.rmtree(dirpath)
 
 
 @contextmanager

--- a/f8a_worker/workers/githuber.py
+++ b/f8a_worker/workers/githuber.py
@@ -118,8 +118,8 @@ class GithubTask(BaseTask):
         gh = github.Github(login_or_token=token)
         try:
             repo = gh.get_repo(full_name_or_id=self._repo_name, lazy=False)
-        except github.GithubException as e:
-            self.log.exception(str(e))
+        except github.GithubException:
+            self.log.error("Failed to get repo %s" % self._repo_name)
             result_data['status'] = 'error'
             return result_data
 

